### PR TITLE
Add UI for Value Tracking

### PR DESCRIPTION
**Pull request by ryzngard**
This adds the start of the UI for value tracking. Builds off of https://github.com/dotnet/roslyn/pull/51898

### CODE REVIEW

File: src/VisualStudio/Core/Def/ValueTracking/ValueTrackingCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+
+            var textSpan = new TextSpan(caretPosition.Value.Position, 0);
+            var sourceTextContainer = args.SubjectBuffer.AsTextContainer();
+            var document = sourceTextContainer.GetOpenDocumentInCurrentContext();
+            if (document is null)
+            {
+                return false;
+            }
+
+            var selectedSymbol = _threadingContext.JoinableTaskFactory.Run(() => GetSelectedSymbolAsync(textSpan, document, cancellationToken));
```
Comment (jmarolf) [inline src/VisualStudio/Core/Def/ValueTracking/ValueTrackingCommandHandler.cs:89]: Have we talked to the platform folks about it being possible to have an async command handler so we don't need to use JTF for this?

File: src/VisualStudio/Core/Def/ValueTracking/ValueTrackingToolWindow.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+namespace Microsoft.VisualStudio.LanguageServices.ValueTracking
+{
+    [Guid(Guids.ValueTrackingToolWindowIdString)]
+    internal class ValueTrackingToolWindow : ToolWindowPane
+    {
+        public static ValueTrackingToolWindow? Instance { get; set; }
+        private readonly ValueTrackingTreeViewModel _viewModel;
+
+        // Needed for VSSDK003
+        // See https://github.com/Microsoft/VSSDK-Analyzers/blob/main/doc/VSSDK003.md
```
Comment (jmarolf) [inline src/VisualStudio/Core/Def/ValueTracking/ValueTrackingToolWindow.cs:26]: We should make sure that we eventually change the tool window to be async. Synchronous tool window construction is surprisingly expensive.

**Review by jmarolf — APPROVED** [review]
:shipit:

File: src/VisualStudio/Core/Def/ValueTracking/ValueTrackingToolWindow.cs (hunk 2 above)
Comment (sharwell) [reply src/VisualStudio/Core/Def/ValueTracking/ValueTrackingToolWindow.cs:26]: Yes, plus async tool windows are not difficult to implement.

**Comment by MartyIX** [comment]
Could anyone tell me what this feature is about? It sounds like something I'm interested in one of my analyzer. Thanks to a kind soul in advance! :)

**Comment by ryzngard** [comment]
@MartyIX this is related to #25994

This isn't the final UI, we're slowly making progress on this feature into a feature branch. Let me know if you have any questions!

**Comment by MartyIX** [comment]
Thanks Andrew! I will check the source code, it looks useful for my scenario :-)

File: src/VisualStudio/Core/Def/ValueTracking/ValueTrackingToolWindow.cs (hunk 2 above)
Comment (ryzngard) [reply src/VisualStudio/Core/Def/ValueTracking/ValueTrackingToolWindow.cs]: I _think_ I fixed this. I was close to having it, didn't realize I should just not block on window show. I followed https://github.com/madskristensen/AsyncToolWindowSample/

File: src/VisualStudio/Core/Def/ValueTracking/ValueTrackingCommandHandler.cs (hunk 1 above)
Comment (ryzngard) [reply src/VisualStudio/Core/Def/ValueTracking/ValueTrackingCommandHandler.cs]: Moved this to just RunAsync, similar to https://github.com/madskristensen/AsyncToolWindowSample/blob/master/src/Commands/ShowToolWindow.cs

**Comment by ryzngard** [comment]
I'm going to merge this, but feel free to leave more feedback and I'll address in a separate PR

File: src/VisualStudio/Core/Def/ValueTracking/ValueTrackingCommandHandler.cs (hunk 1 above)
Comment (jmarolf) [reply src/VisualStudio/Core/Def/ValueTracking/ValueTrackingCommandHandler.cs]: Thanks, I like this a lot better. It being fire-and-forget seems fine compared to blocking the caller on async work.

**Pull request merged**